app/src/main/java/com/andela/helpmebuy/adapters/ReceivedRequestAdapter.java

```diff
@@ -42,30 +42,29 @@ public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
     public void onBindViewHolder(final CustomViewHolder holder, final int position) {
         DateManager dateManager = new DateManager();
         PurchaseRequest request = purchaseRequests.get(position);
-        holder.sender.setText(request.getSendersFullName());
         String items = getItemsDescription(request);
+        String date = dateManager.formatTime(request.getDateMillis());
+        holder.sender.setText(request.getSendersFullName());
         holder.description.setText(items);
-        holder.date.setText(dateManager.formatTime(request.getDateMillis()));
+        holder.date.setText(date);
     }
 
     public String getItemsDescription(PurchaseRequest request){
         String itemName = "";
         ArrayList<PurchaseItem> items = request.getPurchaseList();
-        if(items.size() == 1) {
-            return itemName + items.get(0).getItemName();
+        if(items.size() > 0) {
```

Comment:
It checks each block as long as it does not meet a return statment in any of the if blocks. It would have failed if the subsequent blocks were `else if` blocks.